cc/defs.bzl

```diff
@@ -45,10 +45,22 @@ objc_import = _objc_import
 
 # DEPRECATED: use rule from com_google_protobuf repository
 def cc_proto_library(**kwargs):
+    """Deprecated redirection macro for cc_proto_library.
+
+    Use cc_proto_library from com_google_protobuf.
+
+    On Bazel <8, redirects to native.cc_proto_library.
+    On Bazel >=8, redirects to a mock rule, that fails when analized.
+    This allows for a gradual migration away from this macro.
+
+    Args:
+      **kwargs: passed directly into cc_proto_library
+    """
+    __cc_proto_library = getattr(native, "cc_proto_library", _cc_proto_library)
     if "deprecation" not in kwargs:
-        _cc_proto_library(deprecation = "Use cc_proto_library from com_google_protobuf", **kwargs)
+        __cc_proto_library(deprecation = "Use cc_proto_library from com_google_protobuf", **kwargs)
```

Comment:
This message should include the `.bzl` label and mention the `protobuf` module when Bzlmod is used (https://github.com/bazel-contrib/bazel_features/blob/5d3fd918587fc9c6df1f82b7b0b81f139e5f710b/features.bzl#L64).

Reply:
I added more information to the message. I prefer not to be too smart about it and not emit a different message for bzlmod.